lib/adapters/parsers/grok.js

```diff
@@ -0,0 +1,71 @@
+var base = require('./base');
+var errors = require('../../errors');
+var grok = require('node-grok');
+var Promise = require('bluebird');
+var readline = require('readline');
+var sync = require('sync');
+
+module.exports = base.extend({
+
+    initialize: function(options) {
+        this.pattern = options.pattern || '%{GREEDYDATA:message}';
+    },
+
+    parseStream: function(stream, emit) {
+        var self = this;
+
+        return new Promise(function(resolve, reject) {
+            var buffer = [];
+            var pending = 0;
+            var streamEnd = false;
+
+            function checkDone(streamEnd) {
+                if (buffer.length >= self.limit) {
+                    emit(buffer);
+                    buffer = [];
+                }
+
+                if (pending === 0 && streamEnd) {
+                    emit(buffer);
+                    buffer = [];
+                    resolve();
+                }
+            }
+
+            grok.loadDefault(function(patterns) {
+                var pattern = patterns.createPattern(self.pattern);
+                var liner = readline.createInterface({
+                    input: stream,
+                    terminal: false
+                });
+
+                liner.on('line', function(line) {
+                    pending++;
+                    sync(function() {
```

Comment:
`sync` makes this run synchronously, right, so do you need all that `pending` tracking/logic?